libraries/ms-helper/inc/can_uart.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "can_hw.h"
+#include "status.h"
+#include "uart.h"
+
+// Generic CAN HW <-> UART protocol
+// Requires CAN HW, UART to be initialized
+
+// Process RX'd CAN message from UART
+struct CanUart;
+typedef void (*CanUartRxCb)(const struct CanUart *can_uart, uint32_t id, bool extended,
+                            const uint64_t *data, size_t dlc, void *context);
+
+typedef struct CanUart {
+  UARTPort uart;
+  CanUartRxCb rx_cb;
+  void *context;
+} CanUart;
+
+// Note that the provided UART port will have its RX handler overwritten.
+StatusCode can_uart_init(CanUart *can_uart);
+
+// Overrides the CAN HW RX handler to pass RX'd CAN messages over UART
+StatusCode can_uart_hook_can_hw(CanUart *can_uart);
```

Comment:
Is the idea of having these as a separate functions such that:

* `can_uart_init` is used for setting up the UART RX (if the board is receiving UART messages that have been passed through)
* `can_uart_hook_can_hw` is used for setting up CAN RX (if the board is simply passing through CAN messages)

If so, these names are somewhat confusing IMO.